src/bio/bio_buffer.c

```diff
@@ -617,7 +647,48 @@ dma_map_one(struct bio_desc *biod, struct bio_iov *biov,
 		return rc;
 	}
 add_region:
-	return iod_add_region(biod, chk, chk_pg_idx, off, end);
+	return iod_add_region(biod, chk, chk_pg_idx, off, end,
+			      biov->bi_addr.ba_type);
+}
+
+void
+bio_memcpy(struct bio_desc *biod, uint16_t media, void *media_addr,
+	   void *addr, ssize_t n)
+{
+	struct umem_instance *umem = biod->bd_ctxt->bic_umem;
+
+	if (biod->bd_update && media == DAOS_MEDIA_SCM) {
+		/* NB: pmemobj_tx_commit will drain HW buffer */
+		pmemobj_memcpy(umem->umm_pool, media_addr, addr, n,
+			       PMEMOBJ_F_MEM_NODRAIN);
```

Comment:
from my local testing, pmem_memcpy_persist() provides a higher throughput than pmem_memcpy_nodrain(). We should probably use that instead.

Reply:
Yes, perhaps we'd change it back to persistent copy. I think that our tx commit will drain memory controller anyway, so there could be no much difference. I didn't change it in this patch, you could try change it back to persistent copy to see if it helps.